apps/src/applab/applab.js

```diff
@@ -854,33 +866,53 @@ Applab.render = function() {
   );
 };
 
-Applab.exportApp = function(expoOpts) {
-  Applab.runButtonClick();
-  var html = document.getElementById('divApplab').outerHTML;
-  studioApp().resetButtonClick();
-
-  // TODO: find another way to get this info that doesn't rely on globals.
+/**
+ * Export the project for web or use within Expo.
+ * @param {Object} expoOpts
+ */
+Applab.exportApp = async function(expoOpts) {
   const appName = project.getCurrentName() || 'my-app';
 
-  const {mode, expoSnackId, iconUri, splashImageUri} = expoOpts || {};
-  if (mode === 'expoGenerateApk') {
-    return generateExpoApk(
-      {
+  const {mode} = expoOpts || {};
+  switch (mode) {
+    case 'expoGenerateApk':
+    case 'expoCheckApkBuild':
+    case 'expoCancelApkBuild':
+      return expoInteractWithApk(
+        {
+          ...expoOpts,
+          appName
+        },
+        studioApp().config,
+        Applab.setAndroidExportProps
+      );
+    default:
+      // Run, grab the html from divApplab, then reset:
+      Applab.runButtonClick();
+      var html = document.getElementById('divApplab').outerHTML;
+      studioApp().resetButtonClick();
```

Comment:
This seems like an odd way to get the html. Would Applab.getHtml do what you want?

Reply:
I agree and I was surprised when I saw this. This is existing code from the beginning of export (before my time). I can investigate to see if there are alternate approaches.

Reply:
Unfortunately, this looks like it would be a risky change. We need the actual `divApplab` element in the tree. There is some code in `parseFromLevelHtml()` that executes that seems important. So I don't want to change this right now as export has relied on it for a few years.

Reply:
Sounds good. I've made a Tech Debt task to fix this. https://codedotorg.atlassian.net/browse/STAR-557